test/config-luatest/instance_config_schema_test.lua

```diff
@@ -43,7 +43,7 @@ local function validate_fields(config, record)
                 validate_fields(v1, v.items)
             end
         end
-        if v.type ~= 'record' or not v.enterprise_edition or is_enterprise then
+        if not v.enterprise_edition or is_enterprise then
```

Comment:
Just for the record, it was discussed earlier: https://github.com/tarantool/tarantool/pull/9069#discussion_r1311019517.